### PR TITLE
feat: install protobuf tools into local cache

### LINKED ISSUE #796: remove `protoc` installation from Makefile
remove cumbersome installation of `protoc` and use host os package managers instead

### PR DESCRIPTION
**Pull request by troian**
fixes #796

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -306,49 +335,29 @@ proto-update-deps:
...
-    -o "${BIN}/buf" && \
-	chmod +x "${BIN}/buf"
-
-	touch $@
+.PHONY: proto-tools
+proto-tools: cache-setup $(BUF) $(PROTOC)
 
 tools-clean:
-	rm -f proto-tools-stamp buf-stamp
+	rm -rf .cache
```
Comment (Ropes) [inline Makefile:363]: Since you used it above, any reason not to use the `@rm`?

**Review by boz — APPROVED** [review]
:100: :rocket:

File: Makefile (hunk 1 above)
Comment (boz) [reply Makefile:363]: Please let's not use `@` prefix - I think it's good to see what is happening.

File: Makefile [hunk 2]
```diff
@@ -306,49 +335,29 @@ proto-update-deps:
...
+	@echo "Installing protoc buf cli..."
+	@rm -f $(CACHE_BIN)/buf
+	@curl -sSL \
+		"https://github.com/bufbuild/buf/releases/download/v$(BUF_VERSION)/buf-$(UNAME_OS)-$(UNAME_ARCH)" \
+		-o "$(CACHE_BIN)/buf"
+	@chmod +x "$(CACHE_BIN)/buf"
+
+$(PROTOC):
+	@echo "Installing protoc compiler..."
+	@rm -f $(CACHE_BIN)/protoc
```
Comment (boz) [inline Makefile:352]: this is `$(PROTOC)` or `$@` here.

**Pull request merged**